extension/storage/local/filestorage/client.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestorage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.etcd.io/bbolt"
+)
+
+var defaultBucket = []byte(`default`)
+
+type fileStorageClient struct {
+	db *bbolt.DB
+}
+
+func newClient(file string, timeout time.Duration) (*fileStorageClient, error) {
```

Comment:
Is `file` the file name? File path?

Reply:
What does `timeout` do?

Reply:
`file` is the full file path. I'll rename this accordingly.

`timeout` sets the maximum time to wait when trying to obtain a lock on the file. Without setting this, it can wait indefinitely. I'm not certain that this needs to be configurable, but my thinking is that it shouldn't require a rebuild in order to change it, if it ever proves to be an issue. I've also added a note to the documentation.